src/main/java/com/amazonaws/services/kinesis/clientlibrary/proxies/KinesisProxy.java

```diff
@@ -333,19 +354,73 @@ private ListShardsResult listShards(final String nextToken) {
      */
     @Override
     public Shard getShard(String shardId) {
-        if (this.listOfShardsSinceLastGet.get() == null) {
-            //Update this.listOfShardsSinceLastGet as needed.
-            this.getShardList();
+        if (this.cachedShardMap == null) {
+            synchronized (this) {
+                if (this.cachedShardMap == null) {
```

Comment:
Does it need to be this. ? Can do without it.

Reply:
It's a double checked lock to ensure that if someone gets the lock before us, and loads the shard map we don't reload it.